feat: Introduce Organization Membership Requests (#516)
clerk/clerk-docs: docs/references/javascript/organization/membership-request.mdx

---

> title: Organization Membership Request

Reviewer: The other pages in the JS references will be updated in an upcoming JavaScript Docs Revamp. But, essentially, we are using sentence-casing for headings ( see https://github.com/clerk/clerk-docs/blob/main/STYLEGUIDE.md#15-use-sentence-case-for-titles )

```suggestion
title: Organization membership requests
```

> description: These are all methods on the Organization class that allow you to manage the membership requests for an organization.

Reviewer 2: ```suggestion
description: These are methods on the Organization class that allow you to manage the membership requests for an organization.
```

---

> # Organization Membership Request

Reviewer: ```suggestion
# Organization membership requests
```

> These are all methods on the [Organization](/docs/references/javascript/organization/organization) class that allow you to manage the membership requests for an organization.

Reviewer 2: ```suggestion
These are methods on the [Organization](/docs/references/javascript/organization/organization) class that allow you to manage the membership requests for an organization.
```

## `getMembershipRequests()`

> Retrieves a list of domains for the currently active organization.

Reviewer: ```suggestion
Retrieves a list of membership requests for the currently active organization.
```

```typescript
function getMembershipRequests(params?: GetMembershipRequestParams): Promise<PaginatedResources<OrganizationMembershipRequest>>;
```

### `GetMembershipRequestParams`

| Name | Type | Description |
| --- | --- | --- |

> | `initialPage?` | `number` | A number that can be used to skip the first n-1 pages. For example, if `initialPage` is set to 10, it is will skip the first 9 pages and will fetch the 10th page. |

Reviewer 2: ```suggestion
| `initialPage?` | `number` | The page to fetch results from. For example, if `initialPage` is set to 10, it  will skip the first 9 pages and will fetch the 10th page. |
```

Reviewer: I don't know if I agree. It's not the page to fetch results from, it's the starting page to begin fetching results from.

| `pageSize?` | `number` | A number that indicates the maximum number of results that should be returned for a specific page. |

> | `status?` | `'pending' \| 'accepted' \| 'revoked'` | The status an request can have. |

Reviewer 2: ```suggestion
| `status?` | `'pending' \| 'accepted' \| 'revoked'` | The status a request can have. |
```

### Returns

> | Type | Description |
> | --- | --- |
> | <code>Promise\<[PaginatedResponse][pag-res-ref]\<[OrganizationMembershipRequest][mem-request-ref]\>></code> | This method returns a `Promise` which resolves with a list of [`OrganizationMembershipRequest`][mem-request-ref] objects. |

Reviewer: We are now deduping reference links ( see the [styleguide](https://github.com/clerk/clerk-docs/blob/main/STYLEGUIDE.md#13-de-dupe-reference-links) )

```suggestion
| Type | Description |
| --- | --- |
| <code>Promise\<[PaginatedResponse][pag-res-ref]\<[OrganizationMembershipRequest][mem-request-ref]\>></code> | This method returns a `Promise` which resolves with a list of `OrganizationMembershipRequest` objects. |
```
